src/runner.ts

```diff
@@ -82,6 +81,30 @@ export interface GitHubRunnersProps {
    * @default 10 minutes
    */
   readonly idleTimeout?: cdk.Duration;
+
+  /**
+   * The number of days log events are kept in CloudWatch Logs. When updating
+   * this property, unsetting it doesn't remove the log retention policy. To
+   * remove the retention policy, set the value to `INFINITE`.
+   *
+   * @default logs.RetentionDays.ONE_MONTH
+   */
+  readonly logRetention?: logs.RetentionDays;
+
+
+  /**
+   * Removal policy for logs of image builds. If deployment fails on the custom resource, try setting this to `RemovalPolicy.RETAIN`. This way the CodeBuild logs can still be viewed, and you can see why the build failed.
+   *
+   * We try to not leave anything behind when removed. But sometimes a log staying behind is useful.
+   *
+   * @default RemovalPolicy.DESTROY
+   */
+  readonly logRemovalPolicy?: RemovalPolicy;
+
+  /**
+   * Name of the log group created for the state machine. If not provided, a name will be generated.
```

Comment:
Can we make it optional? Is there a strong use case for always having logs? I find the UI very useful for step functions, but I might be missing something.

Reply:
I don't think you'll get that information from these logs. You will only get the repo name and the labels for the workflow. Failures will only be present in this log if the entire runner failed to start. Whether or not the job succeeded is not visible to the step function.

I want to add metrics for all of those pieces of information you mentioned. See #59.